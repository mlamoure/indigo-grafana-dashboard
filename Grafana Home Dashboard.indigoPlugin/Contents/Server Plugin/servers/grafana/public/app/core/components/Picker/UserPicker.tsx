import React, { Component } from 'react';
import Select from 'react-select';
import PickerOption from './PickerOption';
import { debounce } from 'lodash';
import { getBackendSrv } from 'app/core/services/backend_srv';
import { User } from 'app/types';

export interface Props {
  onSelected: (user: User) => void;
  value?: string;
  className?: string;
}

export interface State {
  isLoading: boolean;
}

export class UserPicker extends Component<Props, State> {
  debouncedSearch: any;

  constructor(props) {
    super(props);
    this.state = { isLoading: false };
    this.search = this.search.bind(this);

    this.debouncedSearch = debounce(this.search, 300, {
      leading: true,
      trailing: true,
    });
  }

  search(query?: string) {
    const backendSrv = getBackendSrv();

    this.setState({ isLoading: true });

    return backendSrv
      .get(`/api/org/users?query=${query}&limit=10`)
      .then(result => {
        return {
          options: result.map(user => ({
            id: user.userId,
            label: `${user.login} - ${user.email}`,
            avatarUrl: user.avatarUrl,
            login: user.login,
          })),
        };
      })
      .finally(() => {
        this.setState({ isLoading: false });
      });
  }

  render() {
    const { value, className } = this.props;
    const { isLoading } = this.state;

    return (
      <div className="user-picker">
        <Select.Async
          valueKey="id"
          multi={false}
          labelKey="label"
          cache={false}
          isLoading={isLoading}
          loadOptions={this.debouncedSearch}
          loadingPlaceholder="Loading..."
          noResultsText="No users found"
          onChange={this.props.onSelected}
          className={`gf-form-input gf-form-input--form-dropdown ${className || ''}`}
          optionComponent={PickerOption}
          placeholder="Select user"
          value={value}
          autosize={true}
        />
      </div>
    );
  }
}
