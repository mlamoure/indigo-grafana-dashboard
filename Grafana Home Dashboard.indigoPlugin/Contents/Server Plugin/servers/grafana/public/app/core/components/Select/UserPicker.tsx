// Libraries
import React, { Component } from 'react';
import _ from 'lodash';

// Components
import { LegacyForms } from '@grafana/ui';
const { AsyncSelect } = LegacyForms;

// Utils & Services
import { debounce } from 'lodash';
import { getBackendSrv } from '@grafana/runtime';

// Types
import { User } from 'app/types';

export interface Props {
  onSelected: (user: User) => void;
  className?: string;
}

export interface State {
  isLoading: boolean;
}

export class UserPicker extends Component<Props, State> {
  debouncedSearch: any;

  constructor(props: Props) {
    super(props);
    this.state = { isLoading: false };
    this.search = this.search.bind(this);

    this.debouncedSearch = debounce(this.search, 300, {
      leading: true,
      trailing: true,
    });
  }

  search(query?: string) {
    this.setState({ isLoading: true });

    if (_.isNil(query)) {
      query = '';
    }

    return getBackendSrv()
      .get(`/api/org/users/lookup?query=${query}&limit=10`)
      .then((result: any) => {
        return result.map((user: any) => ({
          id: user.userId,
          value: user.userId,
          label: user.login,
          imgUrl: user.avatarUrl,
          login: user.login,
        }));
      })
      .finally(() => {
        this.setState({ isLoading: false });
      });
  }

  render() {
    const { className, onSelected } = this.props;
    const { isLoading } = this.state;

    return (
      <div className="user-picker">
        <AsyncSelect
          className={className}
          isLoading={isLoading}
          defaultOptions={true}
          loadOptions={this.debouncedSearch}
          onChange={onSelected}
          placeholder="Select user"
          noOptionsMessage={() => 'No users found'}
        />
      </div>
    );
  }
}
