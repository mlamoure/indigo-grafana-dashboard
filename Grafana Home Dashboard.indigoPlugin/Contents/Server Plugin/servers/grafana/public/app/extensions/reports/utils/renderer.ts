// Get renderer version from its config field, return 1 by default
import { config } from '@grafana/runtime';

export const getRendererMajorVersion = (): number | null => {
  const version = config.rendererVersion;
  if (version === '') {
    return null;
  }

  const semverRegex =
    /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/;
  const res = semverRegex.exec(version);

  let majorVersionStr = '';
  if (res && res.length > 1) {
    majorVersionStr = res[1];
  }

  if (majorVersionStr === '') {
    return 1;
  }

  let majorVersion = parseInt(majorVersionStr, 10);
  if (isNaN(majorVersion)) {
    majorVersion = 1;
  }

  return majorVersion;
};
