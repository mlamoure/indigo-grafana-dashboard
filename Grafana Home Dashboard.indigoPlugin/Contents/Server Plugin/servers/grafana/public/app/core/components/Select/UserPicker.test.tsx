import React from 'react';
// @ts-ignore
import renderer from 'react-test-renderer';
import { UserPicker } from './UserPicker';

jest.mock('@grafana/runtime', () => ({
  getBackendSrv: () => ({ get: jest.fn().mockResolvedValue([]) }),
}));

describe('UserPicker', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<UserPicker onSelected={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
