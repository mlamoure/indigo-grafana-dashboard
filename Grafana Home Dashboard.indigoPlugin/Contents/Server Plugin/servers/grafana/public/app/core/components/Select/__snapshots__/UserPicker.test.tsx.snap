// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPicker renders correctly 1`] = `
<div
  className="user-picker"
>
  <div>
    <div
      className="gf-form-input gf-form-input--form-dropdown css-at6rp9-SelectContainer"
      onKeyDown={[Function]}
    >
      <div
        className="gf-form-select-box__control css-ia584n-Control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className="gf-form-select-box__value-container css-1q9zhbr-ValueContainer"
        >
          <div
            className="gf-form-select-box__placeholder css-d8h0m4-Placeholder"
          >
            Select user
          </div>
          <div
            className="css-zz0hea-Input"
          >
            <div
              className="gf-form-select-box__input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className="gf-form-select-box__indicators css-q46mcr-IndicatorsContainer"
        >
          <div
            className="css-1cvxpvr"
          >
            <svg
              className="css-sr6nr"
              fill="currentColor"
              height={16}
              style={
                Object {
                  "marginTop": "7px",
                }
              }
              viewBox="0 0 24 24"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17,9.17a1,1,0,0,0-1.41,0L12,12.71,8.46,9.17a1,1,0,0,0-1.41,0,1,1,0,0,0,0,1.42l4.24,4.24a1,1,0,0,0,1.42,0L17,10.59A1,1,0,0,0,17,9.17Z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
